packages/dds/tree/src/simple-tree/core/walkSchema.ts

```diff
@@ -69,4 +71,9 @@ export interface SchemaVisitor {
 	 * This includes every field, but also the allowed types array for maps and arrays and the root if starting at {@link walkAllowedTypes}.
 	 */
 	allowedTypes?: (allowedTypes: NormalizedAnnotatedAllowedTypes) => void;
+	/**
+	 * If true, will walk the staged allowed types of the schema in both the node callback and the allowedTypes callback.
+	 * If undefined, will skip any staged allowed types in the node callback but will include them in the allowedTypes callback.
+	 */
+	walkStagedAllowedTypes?: true;
```

Comment:
I think we want something like this, if even to just document that one needs to be concerned about it.

I don't think this option is really enough though.

For example, if you want to walk the portion of the view schema which is enabled (currently supported by the stored schema), passing undefined here might missing some things which have been staged and upgraded already, but passing true might include too much.

Maybe we need an `allowedTypeFilter?: (context: undefined | {node: TreeNodeSchema, field?: string}, type: AnnodatedAllowedType) => boolean` so it would be possible to implement traversing all types currently allowed according to the stored schema.